x509/common.go

```diff
@@ -22,6 +22,11 @@ func readFile(filename string) string {
 	return string(data)
 }
 
+func writeFile(filename, contents string) {
+	err := os.WriteFile(filename, []byte(contents), 0400)
```

Comment:
I suppose writing file atomically is too much in this context (I mean write to tmp file, fsync, and rename).